File: generated_code_examples/f_sharp/regression/xgboost.fs
let score (input : double list) =
    let func0 =
        if input.[12] >= 9.725 then
            if input.[12] >= 19.23 then
                3.5343752
            else
                5.5722494
        else
            if input.[5] >= 6.941 then
                11.1947155
            else
                7.4582143
    let func1 =
        if input.[12] >= 5.1549997 then
            if input.[12] >= 15.0 then
                2.8350503
            else
                4.8024607
        else
            if input.[5] >= 7.406 then
                10.0011215
            else
                6.787523
    0.5 + (func0 + func1)

File: generated_code_examples/elixir/regression/xgboost.ex
defmodule Model do
    @compile {:inline, read: 2}
    defp read(bin, pos) do
        <<_::size(pos)-unit(64)-binary, value::float, _::binary>> = bin
        value
    end
    defp list_to_binary(list) do
        for i <- list, into: <<>>, do: <<i::float>>
    end
    def score(input) do
        input = list_to_binary(input)
        func0 = fn ->
            cond do read(input,12) >= 9.725 ->
                cond do read(input,12) >= 19.23 ->
                    3.5343752
                true ->
                    5.5722494
                end
            true ->
                cond do read(input,5) >= 6.941 ->
                    11.1947155
                true ->
                    7.4582143
                end
            end
        end
        func1 = fn ->
            cond do read(input,12) >= 5.1549997 ->
                cond do read(input,12) >= 15.0 ->
                    2.8350503
                true ->
                    4.8024607
                end
            true ->
                cond do read(input,5) >= 7.406 ->
                    10.0011215
                true ->
                    6.787523
                end
            end
        end
        0.5 + (func0.() + func1.())
    end
end
